caPutLogApp/caPutJsonLogTask.cpp

```diff
@@ -0,0 +1,711 @@
+            fieldVal2Str(reinterpret_cast<char *>(interBuffer), interBufferSize,
+            pold_value, pLogData->type, i);
+            if (pLogData->type == DBF_STRING) {
+                CALL_YAJL_FUNCTION_AND_CHECK_STATUS(status, yajl_gen_string(handle, interBuffer,
+                            strlen(reinterpret_cast<char *>(interBuffer))););
+            } else {
+                CALL_YAJL_FUNCTION_AND_CHECK_STATUS(status, yajl_gen_number(handle,
+                            reinterpret_cast<const char *>(interBuffer),
+                            strlen(reinterpret_cast<char *>(interBuffer))));
+            }
+        }
+    }
+    // Close Json array if we have array value
+    if (pLogData->old_size > 1 && pLogData->type != DBR_CHAR) {
+        CALL_YAJL_FUNCTION_AND_CHECK_STATUS(status, yajl_gen_array_close(handle));
+    }
+    // Add new size, but only if we have array
+    if (pLogData->old_size > 1) {
+        const unsigned char str_oldSize[] = "old-size";
+        CALL_YAJL_FUNCTION_AND_CHECK_STATUS(status, yajl_gen_string(handle, str_oldSize,
+                        strlen(reinterpret_cast<const char*>(str_oldSize))));
+        CALL_YAJL_FUNCTION_AND_CHECK_STATUS(status, yajl_gen_integer(handle, pLogData->old_size));
+    }
+
+    // Add minium and maximum values in case of a burst
+    if (burst && isDbrNumeric(pLogData->type)
+                && pLogData->old_size == 1
+                && pLogData->new_size == 1) {
+        // Add min value
+        const unsigned char str_minVal[] = "min";
+        CALL_YAJL_FUNCTION_AND_CHECK_STATUS(status, yajl_gen_string(handle, str_minVal,
+                        strlen(reinterpret_cast<const char*>(str_minVal))));
+        fieldVal2Str(reinterpret_cast<char *>(interBuffer), interBufferSize,
+                    pmin, pLogData->type, 0);
+        CALL_YAJL_FUNCTION_AND_CHECK_STATUS(status, yajl_gen_number(handle,
+                        reinterpret_cast<const char *>(interBuffer),
+                        strlen(reinterpret_cast<char *>(interBuffer))));
+
+        // Add max value
+        const unsigned char str_maxVal[] = "max";
+        CALL_YAJL_FUNCTION_AND_CHECK_STATUS(status, yajl_gen_string(handle, str_maxVal,
+                        strlen(reinterpret_cast<const char*>(str_maxVal))););
+        fieldVal2Str(reinterpret_cast<char *>(interBuffer), interBufferSize,
+            pmax, pLogData->type, 0);
+        CALL_YAJL_FUNCTION_AND_CHECK_STATUS(status, yajl_gen_number(handle,
+                        reinterpret_cast<const char *>(interBuffer),
+                        strlen(reinterpret_cast<char *>(interBuffer))));
+    }
+
+    /* Close root map */
+    CALL_YAJL_FUNCTION_AND_CHECK_STATUS(status, yajl_gen_map_close(handle));
+
+    /* Get JSON as NULL terminated cstring */
+    const unsigned char * buf;
+    size_t len = 0;
+    yajl_gen_get_buf(handle, &buf, &len);
+
+    /* Get a JSON as a string */
+    std::string json (reinterpret_cast<const char *>(buf));
```

Comment:
Is it intentional that the `logToServer()` method modifies that string (appending a newline), so the `logToPV()` method gets the string with the newline added?

Reply:
Yes, this was a bug. Fixed in: f4683afa4228782f2603c05d58bce7f15055a0cc
The whole purpose why I didn't append new line before was to have log inside a PV without a new line.